spec/commands/overview_spec.rb

```diff
@@ -3,73 +3,116 @@
 require 'commands'
 require 'alces_utils'
 
-RSpec.describe Metalware::Commands::Overview do
+RSpec.shared_context 'mock overview namespaces' do
   include AlcesUtils
-
   let :config_value { 'config_value' }
+  let :static { 'static' }
+  let :fields do
+    [
+      { header: 'heading1', value: static },
+      { header: 'heading2', value: '<%= scope.config.key %>' },
+      { header: 'heading3', value: '' },
+      { header: 'missing-value' },
+      { value: 'missing-header' }
+    ]
+  end
 
   AlcesUtils.mock self, :each do
     ['group1', 'group2', 'group3'].map do |group|
       config(mock_group(group), key: config_value)
     end
   end
+end
 
-  def overview
-    std = AlcesUtils.redirect_std(:stdout) do
+RSpec.describe Metalware::Commands::Overview do
+  include_context 'mock overview namespaces'
+
+  let :name_hash { { header: 'Group Name', value: '<%= group.name %>' } }
+
+  def run_command
+    AlcesUtils.redirect_std(:stdout) do
       Metalware::Utils.run_command(Metalware::Commands::Overview)
     end
-    std[:stdout].read
-  end
-
-  def header
-    overview.lines[1]
   end
 
-  def body
-    overview.lines[3..-2].join("\n")
+  before :each do
+    allow(Metalware::Commands::Overview::Table).to \
+      receive(:new).with(any_args).and_call_original
   end
 
-  it 'includes the group names' do
-    expect(header).to include('Group')
-    alces.groups.each do |group|
-      expect(body).to include(group.name)
-    end
+  def expect_table_with(*inputs)
+    expect(Metalware::Commands::Overview::Table).to \
+      receive(:new).once.with(*inputs).and_call_original
   end
 
-  context 'with a mismatch between no. headers/bodies in overview.yaml' do
-    before :each do
-      Metalware::Data.dump(Metalware::FilePath.overview,
-                           headers: ['I', 'have', '4', 'headers'],
-                           fields: ['I', 'have', '5', 'body', 'parts'])
+  context 'without overview.yaml' do
```

Comment:
I'm not sure there's much advantage of printing (mostly) empty tables when `overview.yaml` is missing, though I guess just listing the groups is somewhat useful. What about printing a warning up front if `overview.yaml` is missing, to inform the user of what they would need to do to get more output shown, as I can imagine people being confused about why more output is not being shown. Something like:

```
warning: `overview.yaml` missing for repo; falling back to minimal `overview` format.
```

Reply:
I am of the opinion if we want to start handling if the the file is missing we should completely commit to validating the file. Currently any structural errors in the `yaml` will create a non-user error which should be caught (an re-raised as a user error).

For the time being, the core functionality has been implemented and I do not want to hold back the `2018.1` release any further.

I have added the validation as a nice thing to have that can be included in the next release.
https://trello.com/c/aKduJydC/250-add-validation-to-overviewyaml

Reply:
I think these are not that closely related and could easily be done independently of each other.

Adding something like
```ruby
unless File.exist? FilePath.overview
  logger.warn '`overview.yaml` missing for repo; falling back to minimal `overview` format.'
end
```
seems a pretty straightforward change which would mitigate an issue I can easily see happening with `overview`, where a user switches the repo they are using and then is confused as to why `overview` is not showing much and not indicating why.

I agree that adding validation of this file seems more involved and for less benefit at this stage.

Reply:
Added a the warning